### PR TITLE
Recommit "add flag to sensor context for first tick after start (#17146)"

### PR DESCRIPTION
**Pull request by prha**
## Summary & Motivation

For sensors that have been turned off for some time, it's useful to have some indication so that user code can decide to limit the range of values to examine (e.g. only look at the last 5 minutes worth of data).

This PR adds the last start time to the stored sensor data, and then adds a flag to the sensor context by comparing the last tick timestamp to the last start timestamp.

[INTERNAL_BRANCH=prha/fix_sensor_grpc]

Test Plan:
BK, TBD

## How I Tested These Changes
BK

### CODE REVIEW

File: python_modules/dagster/dagster/_core/definitions/sensor_definition.py [hunk 1]
```diff
@@ -147,16 +147,20 @@ def __init__(
         sensor_name: Optional[str] = None,
         resources: Optional[Mapping[str, "ResourceDefinition"]] = None,
         definitions: Optional["Definitions"] = None,
+        last_sensor_start_time: Optional[float] = None,
```
Comment (smackesey) [inline python_modules/dagster/dagster/_core/definitions/sensor_definition.py:150]: Need docstring for this

File: python_modules/dagster/dagster/_core/definitions/sensor_definition.py [hunk 2]
```diff
@@ -147,16 +147,20 @@ def __init__(
...
     ):
         from dagster._core.definitions.definitions_class import Definitions
         from dagster._core.definitions.repository_definition import RepositoryDefinition
 
         self._exit_stack = ExitStack()
         self._instance_ref = check.opt_inst_param(instance_ref, "instance_ref", InstanceRef)
-        self._last_completion_time = check.opt_float_param(
+        self._last_tick_completion_time = check.opt_float_param(
```
Comment (smackesey) [inline python_modules/dagster/dagster/_core/definitions/sensor_definition.py:157]: If the property is being renamed to `last_tick_completion_time` seems that the constructor param should be as well

**Review by gibsondan — DISMISSED** [review]
requesting changes for sean's comments (changing the constructor param would be a breaking change though no? I assume that's why?)

**Comment by prha** [comment]
Yeah, changing the constructor param as a breaking change was the main reason I didn't rename... I can add an extra param with default `None` value and then do runtime checks plus deprecation warnings so that we can move off of it in the future though.

**Comment by prha** [comment]
wow, amazing, was not aware.

File: python_modules/dagster/dagster/_core/definitions/multi_asset_sensor_definition.py [hunk 3]
```diff
@@ -196,7 +202,7 @@ class MultiAssetSensorEvaluationContext(SensorEvaluationContext):
             the cursor attribute of SkipReason and RunRequest. Must be a dictionary of asset key
             strings to a stringified tuple of (latest_event_partition, latest_event_storage_id,
             trailing_unconsumed_partitioned_event_ids).
-        last_completion_time (float): DEPRECATED The last time that the sensor was consumed (UTC).
+        last_tick_completion_time (Optional[float]): The last time that the sensor was consumed (UTC).
```
Comment (gibsondan) [inline python_modules/dagster/dagster/_core/definitions/multi_asset_sensor_definition.py:205]: not new here but 'consumed' is a weird verb to use here

**Review by gibsondan — APPROVED** [review]

**Pull request merged**